### PR TITLE
Add bookmark feature to check if user is signedIn

### PR DESCRIPTION
**Pull request by dualcnhq**
**Problem:**
- add access and integrate bookmarks provider inside menu panel
- check if a user is signed in to load the bookmarks list saved 

**Solution:**
- add/link up BookmarkProvider in BookNavigator
- if a user is signed in, load bookmarks saved on a specific page of the book

**Testing Steps:**
- load a book
- access the side menu panel 
- check if Bookmarks option is displayed
- click Bookmarks option
- login layout will be displayed if user is not signed in
- otherwise, it will display the bookmarks list layout

----
_gif layout to follow_

### CODE REVIEW

**Review by iisa — CHANGES_REQUESTED** [review]
hi @dualcnhq this is a solid first pass through the code. it really looks like you have a solid handle as to what is going on.

let's push this abstraction one level up and allow BookNavigator to determine what `<ia-bookmarks>` will display.

primary suggestion: use `this.bookmarksOptions` to determine whether or not <ia-bookmarks> should display given a new field: `displayMode=login|bookmarks` or something like that.

will also look for the unit test[s] in the next round + testing  steps

File: src/BookNavigator/bookmarks/ia-bookmarks.js [hunk 1]
```diff
@@ -444,27 +450,34 @@ class IABookmarks extends LitElement {
...
-          .editedBookmark=${this.editedBookmark}
-          .bookmarks=${{ ...this.bookmarks }}
-          .activeBookmarkID=${this.activeBookmarkID}
-
-          .bookmarkColors=${this.bookmarkColors}
-          .defaultBookmarkColor=${this.defaultColor}
-        ></ia-bookmarks-list>
-        ${this.allowAddingBookmark ? addBookmarkButton : nothing}
-      </section
+        ${this.displayMode === 'login' ? bookmarkLoginLayout : bookmarkSignedInLayout }
```
Comment (nsharma123) [inline src/BookNavigator/bookmarks/ia-bookmarks.js:479]: Can we say these const like `bookmarkLoginLayout` and `bookmarkLoggedInLayout`. login and signedIn seems not correct?

**Comment by iisa** [comment]
Tested on a.o staging - logged out & logged in views are displaying properly.

File: src/BookNavigator/bookmarks/ia-bookmarks.js (hunk 1 above)
Comment (dualcnhq) [reply src/BookNavigator/bookmarks/ia-bookmarks.js]: refactored to be more semantic and readable

**Review by iisa — APPROVED** [review]
worked through side-by-side code review & updated the following:
- using semantic html to display login link
- using getters in <ia-bookmarks> to isolate parts of views
- new component to house login cta `<bookmarks-loginCTA>`
- talked through various best practices & code style
- variable naming - internal component relationships may be different than task scope

**Pull request merged**